File: SharePreferenceNew/app/src/main/java/com/example/sharepreferencenew/DataRepository/DataViewModel.java
package com.example.sharepreferencenew.DataRepository;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class DataViewModel extends ViewModel {
    private MutableLiveData<Float> speed;
    private MutableLiveData<Integer> level;

    public DataViewModel(){
        this.level = new MutableLiveData<>();
        this.speed = new MutableLiveData<>();
    }

    // Level
    public void setLevel(int val){
        this.level.setValue(val);
    }
    public LiveData<Integer> getLevel(){
        return level;
    }


    // speed
    public void setSpeed(Float val){
        this.speed.setValue(val);
    }
    public LiveData<Float> getSpeed(){
        return speed;
    }

}

File: Kotlin-ViewModel-LiveData/app/src/main/java/com/black/xperiments/kotlin_viewmodel_livedata/DataViewModel.kt
package com.black.xperiments.kotlin_viewmodel_livedata

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class DataViewModel:ViewModel() {
    var dataobj: MutableLiveData<DataObj> = MutableLiveData()
    var randomdata: MutableLiveData<Int> = MutableLiveData()
}